networkx/algorithms/lowest_common_ancestors.py

```
@@ -95,41 +115,174 @@ def generate_lca_from_pairs(G, pairs):
+            if len(lca) > 0:
+                yield ((u, v), min(lca, key=key))
+
+    return generate_lca()
+
+
+@not_implemented_for("undirected")
+@nx._dispatchable
+def all_lowest_common_ancestors(G, node1, node2):
+    """Compute all lowest common ancestors for the given pair of nodes.
 
-            if common_ancestors:
-                common_ancestor = next(iter(common_ancestors))
-                while True:
-                    successor = None
-                    for lower_ancestor in G.successors(common_ancestor):
-                        if lower_ancestor in common_ancestors:
-                            successor = lower_ancestor
-                            break
-                    if successor is None:
-                        break
-                    common_ancestor = successor
-                yield ((v, w), common_ancestor)
+    The lowest common ancestor of a pair of nodes is a node that is an ancestor
+    of both nodes and has no descendants that are also ancestors of both nodes.
+    Nodes may have multiple lowest common ancestors in a DAG (Directed Acyclic Graph).
 
-    return generate_lca_from_pairs(G, pairs)
+    Parameters
+    ----------
+    G : NetworkX directed graph
+
+    node1, node2 : nodes in the graph
+
+    Returns
+    -------
+    list
+        A list containing all lowest common ancestors of node1 and node2.
+        Returns an empty list if they have no common ancestors.
+
+    Examples
+    --------
+    >>> G = nx.DiGraph()
+    >>> nx.add_path(G, (0, 1, 2, 3))
+    >>> nx.add_path(G, (0, 4, 3))
+    >>> nx.all_lowest_common_ancestors(G, 2, 4)
+    [0]
+
+    >>> G = nx.DiGraph([(0, 1), (0, 2), (1, 3), (2, 3), (1, 4), (2, 4)])
+    >>> nx.all_lowest_common_ancestors(G, 3, 4)
+    [1, 2]
+
+    See Also
+    --------
+    lowest_common_ancestor, all_pairs_all_lowest_common_ancestors
+    """
+    ans = list(all_pairs_all_lowest_common_ancestors(G, pairs=[(node1, node2)]))
+
+    return ans[0][1]
```

Can this throw index error? If that's the case let's raise an explicit error

---

I believe the shape of `ans` should always be valid. I have added the relevant assertions.